apigw_ca_handler.go

```diff
@@ -0,0 +1,110 @@
+package g8
+
+import (
+	"context"
+	"errors"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/newrelic/go-agent/_integrations/nrlambda"
+	"strings"
+
+	"github.com/JSainsburyPLC/g8/auth"
+
+	"github.com/aws/aws-lambda-go/events"
+	newrelic "github.com/newrelic/go-agent"
+	"github.com/rs/zerolog"
+)
+
+// APIGatewayCustomAuthorizerContext the context for a request for Custom Authorizer
+type APIGatewayCustomAuthorizerContext struct {
+	Context       context.Context
+	Request       events.APIGatewayCustomAuthorizerRequestTypeRequest
+	Response      auth.AuthorizerResponse
+	Logger        zerolog.Logger
+	NewRelicTx    newrelic.Transaction
+	CorrelationID string
+}
+
+// APIGatewayCustomAuthorizerHandlerFunc to populate
+type APIGatewayCustomAuthorizerHandlerFunc func(c *APIGatewayCustomAuthorizerContext) error
+
+// APIGatewayCustomAuthorizerHandler fd
+func APIGatewayCustomAuthorizerHandler(
+	h APIGatewayCustomAuthorizerHandlerFunc,
+	conf HandlerConfig,
+) func(context.Context, events.APIGatewayCustomAuthorizerRequestTypeRequest) (events.APIGatewayCustomAuthorizerResponse, error) {
+
+	return func(ctx context.Context, r events.APIGatewayCustomAuthorizerRequestTypeRequest) (events.APIGatewayCustomAuthorizerResponse, error) {
+		if len(r.MethodArn) == 0 {
```

Comment:
Is this check something that needs to be done here? Should we leave this to the implementor of the lambda authorizer to do themselves? Are there any scenarios where we shouldn't be doing this? Will there ever be in the future?

Reply:
MethodARN is empty when you call lambda from AWS console, there is the button "Test" in the API Gateway. In this case, lambda throws exceptions trying to parse empty ARN. So I added this check to show nice error message, that's it